src/components/utils/useENS.ts

```diff
@@ -0,0 +1,25 @@
+import { ethers } from "ethers";
+import { env } from "../../constants/environmentVariables";
+import { useEffect, useState } from "react";
+
+export function useENS(address: string | null | undefined) {
+  const [ensName, setEnsName] = useState<string | null>(null);
+
+  useEffect(() => {
+    async function resolveENS() {
+      const provider = new ethers.providers.JsonRpcProvider(
```

Comment:
@xnyl can an existing provider be passed in instead? Otherwise [StaticJsonRpcProvider](https://docs.ethers.io/v5/api/providers/jsonrpc-provider/#StaticJsonRpcProvider) could be used to reduce network requests.